sklearn/metrics/_ranking.py

```diff
@@ -1071,20 +1071,22 @@ def label_ranking_average_precision_score(y_true, y_score, *, sample_weight=None
     0.416...
     """
     check_consistent_length(y_true, y_score, sample_weight)
-    y_true = check_array(y_true, ensure_2d=False)
+    y_true = check_array(y_true, ensure_2d=False, accept_sparse="csr")
     y_score = check_array(y_score, ensure_2d=False)
 
     if y_true.shape != y_score.shape:
         raise ValueError("y_true and y_score have different shape")
 
     # Handle badly formatted array and the degenerate case with one label
-    y_type = type_of_target(y_true, input_name="y_true")
-    if y_type != "multilabel-indicator" and not (
-        y_type == "binary" and y_true.ndim == 2
-    ):
-        raise ValueError("{0} format is not supported".format(y_type))
+    if not issparse(y_true):
+        y_type = type_of_target(y_true, input_name="y_true")
+        if y_type != "multilabel-indicator" and not (
+            y_type == "binary" and y_true.ndim == 2
+        ):
+            raise ValueError("{0} format is not supported".format(y_type))
+
+        y_true = csr_matrix(y_true)
```

Comment:
```suggestion
    y_type = type_of_target(y_true, input_name="y_true")
    if y_type != "multilabel-indicator" and not (
        y_type == "binary" and y_true.ndim == 2
    ):
        raise ValueError("{0} format is not supported".format(y_type))

    if not issparse(y_true):
        y_true = csr_matrix(y_true)
```
Those checks should be done not matter if `y_true` is sparse or not. Raising `ValueError` should be covered by a test, see `check_lrap_error_raised`, maybe extend it with CSR input.

Reply:
A test for this is still missing.